src/components/widget/MultiSelect.js

```diff
@@ -0,0 +1,112 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+class MultiSelect extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      data: this.props.options,
+      filter: '',
+      checkedItems: {},
+    };
+  }
+
+  componentDidMount() {
+    this.props.onFocus();
+  }
+
+  static getDerivedStateFromProps(nextProps) {
+    let selected =
+      nextProps.selectedItems && Array.isArray(nextProps.selectedItems)
+        ? nextProps.selectedItems
+        : nextProps.selectedItems !== null
+        ? nextProps.selectedItems.values
+        : null;
+    if (selected !== null) {
+      let updatedCheckedItems = {};
+      selected.map((item) => {
+        if (item) {
+          updatedCheckedItems[item.key] = {
+            key: item.key,
+            caption: item.caption,
+            value: true,
+          };
+        }
+        return item;
+      });
+      return {
+        checkedItems: updatedCheckedItems,
+      };
+    }
+
+    return null;
+  }
+
+  selectItem = (key, caption) => {
```

Comment:
Can you add a comment on what's going on over here ? Why are we transforming `state.checkedItems` to string and then parsing it ?